src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_power_formula.py

```diff
@@ -30,6 +36,26 @@ def generate(  # noqa: DOC502
         )
         grid_successors = self._get_grid_component_successors()
 
+        # Ensure the device has an `ACTIVE_POWER` metric.  When inverters
+        # produce `None` samples, those inverters are excluded from the
+        # calculation by treating their `None` values as `0`s.
+        #
+        # This is not possible for Meters, so when they produce `None`
+        # values, those values get propagated as the output.
+        components = {
+            c
+            for c in grid_successors
+            if c.category
+            in {
+                ComponentCategory.INVERTER,
+                ComponentCategory.EV_CHARGER,
+                ComponentCategory.METER,
+            }
+        }
```

Comment:
I think that comment can go away.  It doesn't make sense after your refactor.

Reply:
Done